api/helpers/oauth/fetch-discord-identity.js

```diff
@@ -0,0 +1,66 @@
+module.exports = {
+  friendlyName: 'Generate Secret',
+
+  description: 'Generates a secret for OAuth state',
+
+  inputs:{
+    token: {
+      type: 'ref',
+      description: 'Discord Access Token',
+      required: true
+    },
+    user: {
+      type: 'ref',
+      description: 'Exisiting User'
+    }
+  },
+
+  fn: async function ({ token, user }, exits) {
+    try {
+      const response = await fetch('https://discord.com/api/users/@me', {
+        headers: {
+          'Authorization': `Bearer ${token.access_token}`,
+        },
+      });
+
+      if (!response.ok) {
+        throw new Error(`Failed to fetch user info: ${response.status}`);
+      }
+
+      const data = await response.json();
+
+      const providerId = data.id;
+      const provider = 'discord';
+      const { email } = data;
+      let { username } = data;
+
+      const prevIdentity = await Identity.findOne({ providerId }).populate('user');
+
+      if (prevIdentity && prevIdentity.user) {
+        if(user && prevIdentity.user.id !== user){
```

Comment:
```suggestion
        if (user && prevIdentity.user.id !== user) {
```

We should add a lint rule around this kind of whitespace